decisions/backlog/0023-error-responses.md

```diff
@@ -0,0 +1,83 @@
+* **RFC6749 (https://www.rfc-editor.org/rfc/rfc6749) Section 4.4:** *If the request failed client authentication or is invalid, the authorization server returns an error response as described in Section 5.2.*
+  * **Section 5.2:** *"The authorization server responds with an HTTP 400 (Bad Request) status code (unless specified otherwise) and includes the following parameters with the response:"*
+    *  "error (REQUIRED)"
+    *  "error_description (OPTIONAL)"
+    *  "error_uri (OPTIONAL)"
+
+For example:
+```javascript
+     HTTP/1.1 400 Bad Request
+     Content-Type: application/json;charset=UTF-8
+     Cache-Control: no-store
+     Pragma: no-cache
+
+     {
+       "error":"..."
+     }   
+```
+
+#### Error Responses
+* **Tech. Specs. (both V1 and V2), Section 6.5/6.9:**
+  * *"Whenever a host system returns an error response, it MUST send a HTTP response such that:"*
+    * *"with response body the error response"*
+  * *"A error response is a JSON object with the following properties:*
+    * *"code: a error response code encoded as a String"*
+    * *"message: a error message encoded as a String"*
+
+For example:
+```javascript
+     HTTP/1.1 400 Bad Request
+     Content-Type: application/json;charset=UTF-8
+     Cache-Control: no-store
+     Pragma: no-cache
+
+     {
+       "code":"...",
+       "message":"..."
+     }   
+```
+
+As shown, these requirements conflict when it comes to the structure of the body of the error response. Therefore, the Tech. Specs. need an update to eliminate this conflict.
+
+## Summary
+
+In general, we propose to stick to standards as much as possible. Hence, we propose to adjust the specifications in the section `Error Responses` (6.5/6.9) so that they do not conflict with a standard. In this case, that means that the specification for the error response can not apply to the `Action Authenticate`. 
+
+
+## Decision
+
+### Technical Specification
+
+**6.9. Error Responses**
+
+Whenever a host system returns an error response for any Action **other** than the `Action Authenticate`, it MUST send a HTTP response such that
+* the HTTP Status Code equals the HTTP Status Code defined for the respective error response code (see Error Codes Table)
+* with content type set to application/json, and
+* with response body the error response
+
+A error response is a JSON object with the following properties:
+* code: a error response code encoded as a String
+* message: a error message encoded as a String
+
+A error response code is a value from column Error Response Code from the table below.
+
+A error message is a human-readable error description. Example values are in column Example Message in the table below.
+
+## Consequences
+
+1. As this ADR is more a correction than a change, the Tech Specs Version number can be updated from 2.0.x to 2.1.y.
```

Comment:
@19-SD - Can you please elaborate on - why this will not be a 2.0.x+1 version after update vs what you are saying - 2.1.y)?

Cc: @raimundo-henriques  @zeitgeist